backend/src/services/pattern_extraction_service.py

```
@@ -0,0 +1,222 @@
+"""
+Pattern Extraction Service
+
+This service handles intelligent pattern extraction from transaction descriptions,
+merchant recognition, and category name suggestions for Phase 4.1.
+"""
+
+import logging
+import re
```

Ruff (0.11.9)

`re` imported but unused

Remove unused import: `re`

(F401)